frontend/src/components/file-explorer/FileExplorer.test.tsx

```diff
@@ -53,47 +59,58 @@ describe("FileExplorer", () => {
       expect(getByText("folder1")).toBeInTheDocument();
       expect(getByText("file1.ts")).toBeInTheDocument();
     });
-    expect(listFiles).toHaveBeenCalledTimes(1); // once for root, once for folder 1
+    expect(listFiles).toHaveBeenCalledTimes(1);
 
-    // The 'await' keyword is required here to avoid a warning during test runs
     await act(async () => {
       await userEvent.click(getByTestId("refresh"));
     });
 
     await waitFor(() => {
-      expect(listFiles).toHaveBeenCalledTimes(2); // 2 from initial render, 2 from refresh button
+      expect(listFiles).toHaveBeenCalledTimes(2);
     });
   });
 
   it("should toggle the explorer visibility when clicking the close button", async () => {
     const { getByTestId, getByText, queryByText } = renderWithProviders(
       <FileExplorer />,
+      {
+        preloadedState: {
+          agent: {
+            curAgentState: AgentState.RUNNING,
+          },
+        },
+      },
     );
 
     await waitFor(() => {
       expect(getByText("folder1")).toBeInTheDocument();
     });
 
-    act(() => {
-      userEvent.click(getByTestId("toggle"));
+    await act(async () => {
+      await userEvent.click(getByTestId("toggle"));
     });
 
-    // it should be hidden rather than removed from the DOM
     expect(queryByText("folder1")).toBeInTheDocument();
     expect(queryByText("folder1")).not.toBeVisible();
   });
 
   it("should upload files", async () => {
     // TODO: Improve this test by passing expected argument to `uploadFiles`
-    const { getByTestId } = renderWithProviders(<FileExplorer />);
+    const { getByTestId } = renderWithProviders(<FileExplorer />, {
+      preloadedState: {
+        agent: {
+          curAgentState: AgentState.RUNNING,
+        },
+      },
+    });
+
     const file = new File([""], "file-name");
     const file2 = new File([""], "file-name-2");
 
-    const uploadFileInput = getByTestId("file-input");
+    const uploadFileInput = await waitFor(() => getByTestId("file-input"));
```

Comment:
NOTE: You could also use `await findByTestId("file-input")` for async tests. `waitFor` uses `findBy*` queries internally, but are usually used for multi-line asserts